resources/voting-controls.js

```diff
@@ -0,0 +1,155 @@
+    $('#id_delete_vote_form_submit_button').show();
+
+    $('#id_vote_button').prop('innerText', translator['change_vote'] + ' (' + voted_points + ')');
+    $('#id_points').prop('value', voted_points);
+    $('#id_note').prop('value', voted_note);
+
+    $('#id_points_error_box').hide();
+    $('#id_note_error_box').hide();
+    has_voted = true;
+}
+
+function deleteVoteUpdate() {
+    $('#id_has_voted_prompt').hide();
+    $('#id_has_not_voted_prompt').show();
+    $('#id_delete_vote_form_submit_button').hide();
+
+    $('#id_vote_button').prop('innerText', translator['vote_points']);
+    $('#id_points').prop('value', '');
+    $('#id_points').prop('placeholder', 10);
+    $('#id_note').prop('value', '');
+    $('#id_note').prop('placeholder', decodeHTMLString(translator['note_justification_help_text']));
+
+    $('#id_points_error_box').hide();
+    $('#id_note_error_box').hide();
+    has_voted = false;
+}
+
+function startVotingControls() {
+    if (has_voted) voteUpdate();
+    else deleteVoteUpdate();
+
+    $('#id_delete_vote_form').on('submit', function (e) {
+        e.preventDefault();
+    });
+
+    $('#id_delete_vote_form_submit_button').on('click', function (e) {
+        e.preventDefault();
+        $.ajax({
+            url: delete_vote_url,
+            type: 'POST',
+            data: $('#id_delete_vote_form').serialize(),
+            success: function () {
+                deleteUserVote(voted_points);
+                voted_points = null;
+                voted_note = null;
+                voted_date = null;
+                deleteVoteUpdate();
+                // Forms are auto disabled to prevent resubmission, but we need to allow resubmission here.
+                $('id_delete_vote_form_submit_button').prop('disabled', false);
+                closeLightbox('id_voting_lightbox');
+            },
+            error: function (data) {
+                alert('Unable to delete vote: ' + data.responseText);
+            },
+        });
+    });
+
+    $('#id_vote_form').on('submit', function (e) {
+        e.preventDefault();
+    });
+
+    $('#id_vote_form_submit_button').on('click', function (e) {
+        e.preventDefault();
+        $.ajax({
+            url: vote_url,
+            type: 'POST',
+            data: $('#id_vote_form').serialize(),
+            success: function (data) {
+                updateUserVote(voted_points, data.points);
+                voted_note = data.note;
+                voted_points = data.points;
+                voted_date = new Date();
+                voteUpdate();
+                // Forms are auto disabled to prevent resubmission, but we need to allow resubmission here.
+                $('#id_vote_form_submit_button').removeAttr('disabled');
+                closeLightbox('id_voting_lightbox');
+            },
+            error: function (data) {
+                let errors = data.responseJSON;
+                if(errors === undefined) {
```

Comment:
```suggestion
                if (errors === undefined) {
```
I swear this is like the 3rd time I've given this suggestion. Please also update the other `if`s...

Reply:
Ok I looked through the JS files and updated the ifs, should be good now....